starter-project-backend-g31/src/__tests__/comments.test.ts

```diff
@@ -0,0 +1,234 @@
+        
+        expect(res.body).not.toBeNull();
+      
+      });
+    });
+    
+    // check for failure case
+    describe('if incorrect route', ()=>{
+        it('return status code 404', async () =>{
+        const res = await request(app).get('/wrongRoute');
+        expect(res.status).toBe(404);
+      });
+    });
+    
+  });
+
+  // test the get comment by id feature
+  describe('GET: "/comment/:id" get a comment by id', ()=>{
+
+    // check succes situations
+    describe('if comment found', ()=>{
+        // check sent data 
+        it('return the comment', async ()=>{
+
+          const res = await request(app).get(`/comment/${commentId}`);
+          
+          expect(res.body).not.toBeNull();
+      
+        });
+        // check status
+        it('return 200 status code', async ()=>{
+
+          const res = await request(app).get(`/comment/${commentId}`);
+
+          expect(res.status).toBe(200);
+        });
+    });
+    
+  });
+
+
+  // check the add comment feature
+  describe('POST: "/comment" Add new comment', ()=>{
+    
+    // if author and description are properly filled
+    describe('given author and description properly', ()=>{
+      // dummy data to insert
+      const dummy = {
+          author: "another person",
+          description: "another description"
+      };
+      // check status code
+      it('status code should be 200', async ()=>{
+        const res = await request(app).post('/comment').send(dummy);
+
+        expect(res.status).toBe(200);
+      });
+
+      // check reeponse data
+      it('response data should be same with dummy', async ()=>{
+        const res = await request(app).post('/comment').send(dummy);
+        
+        // expect(res.body).not.toBeNull();
+        expect(
+          {
+            author: res.body.author,
+            description: res.body.description 
+          }).toStrictEqual(dummy);
+          
+        // expect(res.body.description).toBe(dummy.description);
+      });
+      
+    });
+
+    // if author and description are not filled properly
+    describe('wrong author and description input', ()=>{
+      
+      // check status
+      it('return status code 404', async()=>{
+        const res = await request(app).post('/comment').send({});
+        expect(res.status).toBe(404);
+      });
+      
+    });
+
+  });
+
+
+  //check the update comment by id feature
+  describe('PATCH: "/comment/:id" update existing comment by id', ()=>{
+    describe('if correct id',()=>{
+      
+      // check status
+      it('return status code 200', async ()=>{
+        const res = await request(app).patch(`/comment/${commentId}`).send({
+          author: "changed"
+        });
+
+        expect(res.status).toBe(200);
+      });
+      // check returned content type
+      it('return content type should be json', async()=>{
+        const res = await request(app).patch(`/comment/${commentId}`).send({
+          author: "changed"
+        });

```

Comment:
you can use one block 'it' block and set sequence of it blocks, to check status code, return type and whether it's updated or not.

Reply:
and better to use one describe block for one test scenario and add test suits under it blocks,